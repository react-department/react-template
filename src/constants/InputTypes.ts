enum InputTypes {
  Text = 'TEXT',
  Password = 'PASSWORD',
}

export default InputTypes;
