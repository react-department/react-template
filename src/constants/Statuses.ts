enum Statuses {
  idle = 'idle',
  loading = 'loading',
  failed = 'failed',
}

export default Statuses;
