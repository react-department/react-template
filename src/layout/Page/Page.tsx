import { Outlet } from 'react-router-dom';

import Title from '../../components/Title/Title';
import usePageTitle from '../../hooks/usePageTitle';
import Header from '../Header/Header';

import type { ReactElement } from 'react';

/**
 * Main page layout
 */
function Page(): ReactElement {
  const pageTitle = usePageTitle();
  return (
    <div className="page">
      <Header />
      <Title text={pageTitle} />
      <Outlet />
    </div>
  );
}

export default Page;
