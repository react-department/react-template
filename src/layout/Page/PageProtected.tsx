import { Navigate } from 'react-router-dom';

import { useAppSelector } from '../../store/hooks/useApp';
import { selectUserId } from '../../store/slices/auth/selectors';
import Page from './Page';

import type { ReactElement } from 'react';
import type IPageProtected from './interfaces/IPageProtected';

/**
 * Protected page layout
 */
function PageProtected({ redirectUri }: IPageProtected): ReactElement {
  const id = useAppSelector(selectUserId);
  return id ? (
    <Page />
  ) : (
    <Navigate to={redirectUri} />
  );
}

export default PageProtected;
