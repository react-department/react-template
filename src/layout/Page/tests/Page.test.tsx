import { screen } from '@testing-library/react';

import { render } from '../../../../__tests__/test-utils';
import { testIdHeader } from '../../../constants/TestId';
import Page from '../Page';

describe('<Page />', () => {
  test('should render component', () => {
    const { asFragment } = render(<Page />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('should render component with header and title', () => {
    render(<Page />);
    const header = screen.getByTestId(testIdHeader);
    const h1List = screen.getAllByRole('heading', { level: 1 });
    expect(header).toBeInTheDocument();
    expect(h1List.length).toBeLessThan(2);
  });
});
