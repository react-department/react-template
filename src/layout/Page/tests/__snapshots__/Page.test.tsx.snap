// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> should render component 1`] = `
<DocumentFragment>
  <div
    class="page"
  >
    <header
      class="header"
      data-testid="header"
    >
      <h2
        class="appSubtitle"
        data-testid="subtitle"
      >
        Try to build something great
      </h2>
      <nav
        class="headerNav"
      >
        <a
          class="headerNavLink"
          href="/"
        >
          Home
        </a>
        <a
          class="headerNavLink"
          href="/about"
        >
          About
        </a>
      </nav>
      <div
        class="toggleLanguage"
      >
        <span>
          <a
            class="linkButton"
            data-testid="loginButton"
            href="/login"
          >
            Login
          </a>
        </span>
        <button
          data-testid="toggleLanguageButton"
          type="button"
        >
          EN
        </button>
      </div>
    </header>
    <h1
      class="appTitle"
      data-testid="mainTitle"
    >
      Home
    </h1>
  </div>
</DocumentFragment>
`;
